### PR TITLE
Angular: Fix NgRx Action console logging in dev mode

### PR DESCRIPTION
**Pull request by davidsoergel**
Console logging for NgRx actions was not working because of file loading order issues.  Basically (I think) our `const metaReducers = ...` was not actually set at the time that the AppModule @NgModule decorator was evaluated.

While I was at it, I swapped in ngrx-store-logger in place of our manual logger, for nice color coding and such.

### CODE REVIEW

**Comment by davidsoergel** [comment]
Sorry for the crufty commit history-- I think it's not worth cleaning up in this case since the actual changes are tiny and easy to see, but I'll try to arrange my branches more cleanly in the future :)

File: tensorboard/webapp/reducer_config.ts [hunk 1]
```diff
@@ -12,32 +12,32 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
...
-
-    return result;
-  };
+export function logger(reducer: ActionReducer<State>): ActionReducer<State> {
+  // We must call isDevMode() within this function, not at file load time,
+  // because of load ordering issues vs. the config/bootstrap setup.
+
+  // TODO(stephanwlee, davidsoergel): Why is this called twice?
+  console.log(
+    isDevMode() ? 'DEV mode: Logger enabled' : 'PROD mode: Logger disabled'
```
Comment (psybuzz) [inline tensorboard/webapp/reducer_config.ts]: nit: in Prod, I think we don't gain much by seeing 'Logger disabled'.

File: tensorboard/webapp/reducer_config.ts [hunk 2]
```diff
@@ -12,32 +12,32 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
...
-    console.log('next state', result);
-    console.groupEnd();
-
-    return result;
-  };
+export function logger(reducer: ActionReducer<State>): ActionReducer<State> {
+  // We must call isDevMode() within this function, not at file load time,
+  // because of load ordering issues vs. the config/bootstrap setup.
+
+  // TODO(stephanwlee, davidsoergel): Why is this called twice?
```
Comment (psybuzz) [inline tensorboard/webapp/reducer_config.ts]: I'm curious as well.  It seems ngrx calls metareducers once when creating Reducers, and again in 'updateReducers', called when it actually adds features/reducers to the store.

Maybe we can just do this console.log outside of this function, in the module itself?

File: tensorboard/webapp/reducer_config.ts [hunk 3]
```diff
@@ -12,32 +12,32 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
...
+  console.log(
+    isDevMode() ? 'DEV mode: Logger enabled' : 'PROD mode: Logger disabled'
+  );
+  return isDevMode() ? storeLogger()(reducer) : reducer;
 }
 
-export const metaReducers: MetaReducer<any>[] =
-  config.env === 'dev' ? [logger] : [];
+// TODO(stephanwlee, davidsoergel): consider injecting metareducers via
+// https://ngrx.io/guide/store/recipes/injecting#injecting-meta-reducers
```
Comment (psybuzz) [inline tensorboard/webapp/reducer_config.ts]: Both ways sound reasonable, are there benefits to this approach?

File: tensorboard/webapp/reducer_config.ts (hunk 1 above)
Comment (caisq) [reply tensorboard/webapp/reducer_config.ts]: +1.

Also, I think "Logger" is pretty vague. Can we be more specific in the log message? I.e,

```typescript
if (isDevMod()) {
  console.log('DEV mode: NgRx Store Logger enabled');
}
```

Comment (caisq) [inline tensorboard/webapp/reducer_config.ts]: nit: `isDevMod()` is called twice here. Can we just call it once and use the return value twice?

**Comment by stephanwlee** [comment]
This is fixed by recent changes to the dev environment and what not. Thanks a lot for PR, David!

**Pull request closed without merging**